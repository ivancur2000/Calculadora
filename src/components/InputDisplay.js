import React from 'react';
import PropTypes from 'prop-types';

export const InputDisplay = ({ value }) =>{
    return(
        <input
            type="text"
            className='form-control'
            value={value.join('')}
            disabled
        />
    );
}
InputDisplay.propTypes = {
    value: PropTypes.array.isRequired,
}