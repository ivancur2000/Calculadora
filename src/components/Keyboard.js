import React from 'react';
import PropTypes from 'prop-types';

export const Keyboard = ({ handleOnClickNumber, handleOnClickOperation, handleOnReset, handleOnResolve })=>{

    const btnNumber = [1,2,3,4,5,6,7,8,9,0];
    const operations = ['+', '-', '/', '*'];

    return(
    <div className='flex flex-wrap'>
        <div className='row w-100'>
            <div className='col-6 number__pad'>
                {
                    btnNumber.map(number =>(
                        <button
                            className='btn btn-lg btn-success'
                            key={number}
                            onClick={ ()=> handleOnClickNumber( number ) }
                        >
                            {number}
                        </button>
                    ))
                }
            </div>
            <div className='col-6 operation__pad'>
                <button 
                    className='btn btn-lg btn-warning'
                    onClick={ handleOnReset }
                >
                    CE
                </button>
                <button 
                    className='btn btn-lg btn-danger'
                    onClick={ handleOnResolve }
                >
                    =
                </button>
                {
                    operations.map(symbol => (
                        <button
                            className='btn btn-lg btn-light'
                            key={symbol}
                            onClick={ ()=> handleOnClickOperation(symbol) }
                        >
                            {symbol}
                        </button>
                    ))
                }
            </div>
        </div>
    </div>
    );
}
Keyboard.propTypes = {
    handleOnClickNumber: PropTypes.func.isRequired,
    handleOnClickOperation: PropTypes.func.isRequired,
    handleOnReset: PropTypes.func.isRequired,
    handleOnResolve: PropTypes.func.isRequired,
}