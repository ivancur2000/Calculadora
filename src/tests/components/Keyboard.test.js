import { shallow } from "enzyme";
import { Keyboard } from "../../components/Keyboard";

describe('Pruebas en <Keyboard />', () => {

    const handleOnClickNumber = jest.fn();
    const handleOnClickOperation = jest.fn();
    const handleOnReset = jest.fn();
    const handleOnResolve = jest.fn();
    const wrapper = shallow(
        <Keyboard 
            handleOnClickNumber={handleOnClickNumber}
            handleOnReset={handleOnReset}
            handleOnClickOperation={handleOnClickOperation}
            handleOnResolve={handleOnResolve}
        />
    );
    const btn = wrapper.find('button');

    test('Debe renderizar <Keyboard /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('Debe mostrar 14 botones', () => {
       expect(btn.length).toBe(16); 
    });
    test('Debe ejecutar las funciones al presionar los botones', () => {
        for (let i = 0; i <= 15; i++) {
            btn.at(i).simulate('click');
        }
        expect(handleOnClickNumber).toHaveBeenCalled();
        expect(handleOnClickOperation).toHaveBeenCalled();
        expect(handleOnReset).toHaveBeenCalled();
        expect(handleOnResolve).toHaveBeenCalled();
    });
});