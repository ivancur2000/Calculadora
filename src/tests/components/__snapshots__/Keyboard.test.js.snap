// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Keyboard /> Debe renderizar <Keyboard /> correctamente 1`] = `
<div
  className="flex flex-wrap"
>
  <div
    className="row w-100"
  >
    <div
      className="col-6 number__pad"
    >
      <button
        className="btn btn-lg btn-success"
        key="1"
        onClick={[Function]}
      >
        1
      </button>
      <button
        className="btn btn-lg btn-success"
        key="2"
        onClick={[Function]}
      >
        2
      </button>
      <button
        className="btn btn-lg btn-success"
        key="3"
        onClick={[Function]}
      >
        3
      </button>
      <button
        className="btn btn-lg btn-success"
        key="4"
        onClick={[Function]}
      >
        4
      </button>
      <button
        className="btn btn-lg btn-success"
        key="5"
        onClick={[Function]}
      >
        5
      </button>
      <button
        className="btn btn-lg btn-success"
        key="6"
        onClick={[Function]}
      >
        6
      </button>
      <button
        className="btn btn-lg btn-success"
        key="7"
        onClick={[Function]}
      >
        7
      </button>
      <button
        className="btn btn-lg btn-success"
        key="8"
        onClick={[Function]}
      >
        8
      </button>
      <button
        className="btn btn-lg btn-success"
        key="9"
        onClick={[Function]}
      >
        9
      </button>
      <button
        className="btn btn-lg btn-success"
        key="0"
        onClick={[Function]}
      >
        0
      </button>
    </div>
    <div
      className="col-6 operation__pad"
    >
      <button
        className="btn btn-lg btn-warning"
        onClick={[MockFunction]}
      >
        CE
      </button>
      <button
        className="btn btn-lg btn-danger"
        onClick={[MockFunction]}
      >
        =
      </button>
      <button
        className="btn btn-lg btn-light"
        key="+"
        onClick={[Function]}
      >
        +
      </button>
      <button
        className="btn btn-lg btn-light"
        key="-"
        onClick={[Function]}
      >
        -
      </button>
      <button
        className="btn btn-lg btn-light"
        key="/"
        onClick={[Function]}
      >
        /
      </button>
      <button
        className="btn btn-lg btn-light"
        key="*"
        onClick={[Function]}
      >
        *
      </button>
    </div>
  </div>
</div>
`;
